IHP/IDE/SchemaDesigner/Parser.hs

```diff
@@ -194,6 +212,25 @@ sqlType = choice
                     try (symbol' "TIME")
                     pure PTime
 
+                numericPS = do
+                    try (symbol' "NUMERIC(")
+                    values <- between (space) (char ')' >> space) (varExpr `sepBy` (char ',' >> space))
+                    case values of
+                        [VarExpression precision, VarExpression scale] -> pure (PNumericP (tryTextToInt precision) (tryTextToInt scale))
+                        [VarExpression precision] -> pure (PNumericP (tryTextToInt precision) 0)
+                        _ -> pure PNumeric
+
+                numeric = do
+                    try (symbol' "NUMERIC")
+                    pure PNumeric
+
+                varyingN = do
+                    try (symbol' "VARYING(") <|> try (symbol' "VARCHAR(") <|> try (symbol' "CHAR(") <|> try (symbol' "CHARACTER(")
```

Comment:
Isn't `character` different from `varchar`? This would turn any `CHARACTER(1)` to `VARCHAR(1)` when parsing and then resaving the schema.